react/CozyDialogs/SpecificDialogs/ShortcutDialog.jsx

```diff
@@ -0,0 +1,159 @@
+import PropTypes from 'prop-types'
+import React, { useState, forwardRef } from 'react'
+
+import withSpecificDialogsLocales from './withSpecificDialogsLocales'
+import { ConfirmDialog } from '..'
+import Button from '../../Buttons'
+import Stack from '../../Stack'
+import TextField from '../../TextField'
+import { useAlert } from '../../providers/Alert'
+import { useI18n } from '../../providers/I18n'
+
+const ENTER_KEY = 13
+
+const isURLValid = url => {
+  try {
+    new URL(url)
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+const makeURLValid = str => {
+  if (isURLValid(str)) return str
+  else if (isURLValid(`https://${str}`)) return `https://${str}`
+  return false
+}
+
+const makeFilenameValid = fileName =>
+  fileName.endsWith('.url') ? fileName : fileName + '.url'
+
+const makeFilenameHumanReadable = fileName =>
+  fileName.endsWith('.url') ? fileName.slice(0, -4) : fileName
+
+const ShortcutDialog = forwardRef(({ shortcut, onSave, onClose }, ref) => {
+  const { t } = useI18n()
+  const { showAlert } = useAlert()
+
+  const initialName = makeFilenameHumanReadable(shortcut?.name || '')
+  const initialUrl = shortcut?.url || ''
+  const isEditing = !!shortcut
+
+  const [fileName, setFilename] = useState(initialName)
+  const [url, setUrl] = useState(initialUrl)
+
+  const handleKeyDown = e => {
+    if (e.keyCode === ENTER_KEY) {
+      createShortcut()
+    }
+  }
+
+  const createShortcut = async () => {
```

Comment:
We can extract this as well. The benefit of that we can see how much dep there is, it's a good information about responsibility I think